src/Tribe/Views/V2/View_Register.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * The View registration facade.
+ *
+ * @package Tribe\Events\Views\V2
+ * @since   TBD
+ */
+
+namespace Tribe\Events\Views\V2;
+
+/**
+ * Class View_Register
+ *
+ * @package Tribe\Events\Views\V2
+ * @since   TBD
+ */
+class View_Register {
+	/**
+	 * Slug for the view.
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * Name for the view.
+	 *
+	 * @var string
+	 */
+	protected $name;
+
+	/**
+	 * Class name for the view.
+	 *
+	 * @var string
+	 */
+	protected $class;
+
+	/**
+	 * Priority order for the view registration.
+	 *
+	 * @var int
+	 */
+	protected $priority;
+
+	/**
+	 * View_Register constructor.
+	 *
+	 * @param string $slug Slug for the view.
+	 * @param string $name Name for the view.
+	 * @param string $class Class name for the view.
+	 * @param int $priority Priority order for the view registration.
+	 */
+	public function __construct( $slug, $name, $class, $priority = 40 ) {
+		$this->slug     = $slug;
+		$this->name     = $name;
+		$this->class    = $class;
+		$this->priority = $priority;
+
+		$this->add_actions();
+		$this->add_filters();
+
+		$asset_registration_object = call_user_func( $this->class . '::get_asset_origin', $this->slug );
+		call_user_func( $this->class . '::register_assets', $asset_registration_object );
+	}
+
+	/**
+	 * Adds actions for view registration.
+	 *
+	 * @since TBD
+	 */
+	protected function add_actions() {
+		add_action( 'tribe_events_pre_rewrite', [ $this, 'filter_add_routes' ], 5 );
+	}
+
+	/**
+	 * Adds filters for view registration.
+	 *
+	 * @since TBD
+	 */
+	protected function add_filters() {
+		add_filter( 'tribe_events_views', [ $this, 'filter_events_views' ] );
+		add_filter( 'tribe-events-bar-views', [ $this, 'filter_tec_bar_views' ], $this->priority, 1 );
```

Comment:
Do we still use this filter for something on V2? That seems like a v1 filter.

Reply:
Brought this back. It is what is used to power the settings page for Views.